### PR TITLE
Psy drain, another way of getting psych points and larva

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

All xenos gain the psy drain ability, wich can be executed on dead marine bodies. Every dead marine psy drained will give 60 pysh points, the equivalent of 2 minutes of full generator mining.

A psy drained marine is NOT perma dead, it's not even damaging him.

Silo price ranked up to 1200 from 900 to take into account this new way of getting psych points

EDIT: psy drain will also give 1/8 of a larva, on top of the psych points

## Why It's Good For The Game

Give xenos a reward to kill marines, and make them deserve their psych points. This is currently not replacing the generators psych generation, but i'll maybe nerf the generators later so psych points are aquired more actively

Also gives larva point now, because i saw some hive prefering having no silos rather than having to defend them. I think it's a neat choice, and this ability give xenos more possible strategies to experiment

## Changelog
:cl:
add: Psy drain, an ability that all xenos have that allow them to get psych points and 1/8th of a larva from dead marines. A psy drained marine is NOT perma dead, this is not headbite
balance: Silos get more expensive, to account for psy drain
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 1]
```diff
@@ -1198,3 +1198,81 @@
...
+		return FALSE
+	succeed_activate() //dew it
+
+/datum/action/xeno_action/activable/psydrain/use_ability(mob/M)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/carbon/victim = M
+
+	if(HAS_TRAIT(victim, TRAIT_PSY_DRAINED))
+		to_chat(X, "<span class='warning'>Someone drained the life force of our victim before we could do it!</span>")
+		return
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1264]: fail activate

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Review by TiviPlus — APPROVED** [review]

**Review by TiviPlus — APPROVED** [review]

**Comment by Swagile** [comment]
so this is hilariously broken right now because xenos can get 9 silos and 10 + burrowed before shutters open on maps like prison due to colonist dead body psy draining

**Comment by Losenis** [comment]
yep
image

File: code/game/data_huds.dm [hunk 2]
```diff
@@ -216,7 +216,9 @@
 	switch(stat)
 		if(DEAD)
 			simple_status_hud.icon_state = ""
-			infection_hud.icon_state = "huddead" //Xenos sense dead hosts, and no longer their larvas inside, which fall into stasis and no longer grow.
+			infection_hud.icon_state = "psy_drain"
+			if(HAS_TRAIT(src, TRAIT_PSY_DRAINED))
+				infection_hud.icon_state = "huddead"
```
Comment (TiviPlus) [inline code/game/data_huds.dm:221]: This isnt right

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 3]
```diff
@@ -1198,3 +1198,83 @@
...
+		to_chat(X, "<span class='warning'>Someone drained the life force of our victim before we could do it!</span>")
+		return fail_activate()
+	
+	playsound(X, 'sound/magic/end_of_psy_drain.ogg', 40)
+
+	X.visible_message("<span class='xenodanger'>\The [victim]'s life force is drained by \the [X]!</span>", \
+	"<span class='xenodanger'>We suddenly feel \the [victim]'s life force streaming into us!</span>")
+
+	victim.do_jitter_animation(2)
+
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:1272]: Could also do filter stuff here if you like

Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/abilities.dm:1272]: yeah, I asked ciel to do me one

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**